edx_event_bus_kafka/publishing/event_producer.py

```diff
@@ -0,0 +1,221 @@
+        return None
+
+    if bootstrap_servers := getattr(settings, 'KAFKA_BOOTSTRAP_SERVERS', None):
+        producer_settings = {
+            'bootstrap.servers': bootstrap_servers,
+        }
+    else:
+        warnings.warn("Cannot configure event-bus-kafka: Missing setting KAFKA_BOOTSTRAP_SERVERS")
+        return None
+
+    if getattr(settings, 'KAFKA_API_KEY', None) and getattr(settings, 'KAFKA_API_SECRET', None):
+        producer_settings.update({
+            'sasl.mechanism': 'PLAIN',
+            'security.protocol': 'SASL_SSL',
+            'sasl.username': settings.KAFKA_API_KEY,
+            'sasl.password': settings.KAFKA_API_SECRET,
+        })
+
+    schema_registry_client = SchemaRegistryClient(schema_registry_config)
+    signal_serializer = get_serializer(signal)
+
+    def inner_to_dict(event_data, ctx=None):  # pylint: disable=unused-argument
+        """Tells Avro how to turn objects into dictionaries."""
+        return signal_serializer.to_dict(event_data)
+
+    # Serializers for key and value components of Kafka event
+    key_serializer = AvroSerializer(
+        schema_str=extract_key_schema(signal_serializer, event_key_field),
+        schema_registry_client=schema_registry_client,
+        to_dict=inner_to_dict,
+    )
+    value_serializer = AvroSerializer(
+        schema_str=signal_serializer.schema_string(),
+        schema_registry_client=schema_registry_client,
+        to_dict=inner_to_dict,
+    )
+
+    producer_settings.update({
+        'key.serializer': key_serializer,
+        'value.serializer': value_serializer,
+    })
+
+    return SerializingProducer(producer_settings)
+
+
+def on_event_deliver(err, evt):
+    """
+    Simple callback method for debugging event production
+
+    Arguments:
+        err: Error if event production failed
+        evt: Event that was delivered (or failed to be delivered)
+    """
+    if err is not None:
+        logger.warning(f"Event delivery failed: {err!r}")
+    else:
+        # Don't log msg.value() because it may contain userids and/or emails
+        logger.info(f"Event delivered to topic {evt.topic()}; key={evt.key()}; "
+                    f"partition={evt.partition()}")
+
+
+def send_to_event_bus(signal: OpenEdxPublicSignal, topic: str, event_key_field: str, event_data: dict) -> None:
+    """
+    Send a signal event to the event bus under the specified topic.
+
+    If the Kafka settings are missing or invalid, return with a warning.
+
+    Arguments:
+        signal: The original OpenEdxPublicSignal the event was sent to
+        topic: The event bus topic for the event
+        event_key_field: Path to the event data field to use as the event key (period-delimited
+          string naming the dictionary keys to descend)
+        event_data: The event data (kwargs) sent to the signal
+    """
+    producer = get_producer_for_signal(signal, event_key_field)
+    if not producer:
+        return
```

Comment:
Good thing I did some manual testing! Turns out that SerializingProducer has a truthiness of False. D-:

Reply:
...because it has a `__len__` method, which can return 0. yay.